### PR TITLE
ci: only publish after releases are created

### LINKED ISSUE #2020: [ci] newly created packages are published before the release PR is merged
There seems to be a bug in our release pipeline where newly created packages are published before a release PR is merged. We should guard [in our workflow](https://github.com/open-telemetry/opentelemetry-js-contrib/blame/9df30ea0fd822a69203b818b8fbe34e1e1c8bced/.github/workflows/release-please.yml#L66-L72) so that it only releases on commits that are created by a release PR.

I've noticed this as the `@opentelemetry/winston-transport` package was published without an associated release PR.

**Resources:**
- Job that published the package: https://github.com/open-telemetry/opentelemetry-js-contrib/actions/runs/8286523704/job/22676722070
- Publish step which should only run after a release PR was merged: https://github.com/open-telemetry/opentelemetry-js-contrib/blame/9df30ea0fd822a69203b818b8fbe34e1e1c8bced/.github/workflows/release-please.yml#L66-L72

### PR DESCRIPTION
**Pull request by JamieDanielson**
## Which problem is this PR solving?

- closes #2020

## Short description of the changes

- Add conditional checks to steps that require a release to be created first.

According to [`release-please-action` Outputs section](https://github.com/google-github-actions/release-please-action?tab=readme-ov-file#outputs) `releases_created` is `true` if any release was created, `false` otherwise. This should be the right option compared to the previous attempt that used `release_created` which is only `true` if a root component release was created, `false` otherwise. Thanks for the [tip](https://github.com/open-telemetry/opentelemetry-js-contrib/issues/2020#issuecomment-2078830357) @pichlermarc ! Hopefully this one works better. 🤞

### CODE REVIEW

File: .github/workflows/release-please.yml [hunk 1]
```diff
@@ -54,11 +54,15 @@ jobs:
...
       # get main again
       - name: Checkout Repository
+        # only checkout if a release has been created
+        if: ${{ steps.release.outputs.releases_created }}
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
       - name: Rebuild Packages
+        # only checkout if a release has been created
```
Comment (blumamir) [inline .github/workflows/release-please.yml:64]: nit: update the comment copied from the previous step or remove it

**Review by blumamir — APPROVED** [review]

**Review by david-luna — APPROVED** [review]

File: .github/workflows/release-please.yml (hunk 1 above)
Comment (pichlermarc) [inline .github/workflows/release-please.yml:64]:
```suggestion
        # only rebuild if a release has been created
```

File: .github/workflows/release-please.yml [hunk 2]
```diff
@@ -67,6 +71,8 @@ jobs:
       # need to publish all unpublished versions to npm here
       # See: https://github.com/lerna/lerna/tree/main/commands/publish#bump-from-package
       - name: Publish to npm
+        # only checkout if a release has been created
```
Comment (pichlermarc) [inline .github/workflows/release-please.yml:74]:
```suggestion
        # only publish if a release has been created
```

File: .github/workflows/release-please.yml (hunk 1 above)
Comment (pichlermarc) [reply .github/workflows/release-please.yml]: I applied the suggestion via a comment

Comment (pichlermarc) [reply .github/workflows/release-please.yml]: see https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2160/commits/2acfea62a2075580c2535a3567b56a5e85c2ccb5 and https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2160/commits/fe6da36260685c499a5073dc2c205d713768b622 :slightly_smiling_face:

Comment (JamieDanielson) [reply .github/workflows/release-please.yml]: Oops good eye thanks both!

**Pull request merged**